### PR TITLE
Add WildFire Dataset object and split strategies

### PR DESCRIPTION
**Pull request by x0s**
Hello,

This PR aims at:
- Add `WildFireDataset`  pytorch compatible class
- Add Strategy to split the dataset in train/val/test without mixing the fire_ids

Example:
```python
from torch.utils.data import DataLoader
from torchvision.transforms import transforms
from pyronear.datasets.wildfire import (WildFireDataset,
                                        WildFireSplitter)

wildfire = WildFireDataset(metadata='wildfire.csv',
                           target_names=['fire', 'clf_confidence'],
                           path_to_frames=path_to_frames)

ratios = {'train': 0.7, 'val': 0.15, 'test':0.15}
transforms = {'train': transforms.RandomCrop(10), 'val': None, 'test': None}

splitter = WildFireSplitter(ratios, transforms)
splitter.fit(wildfire)

wildfire_loader_train = DataLoader(splitter.train, batch_size=64, shuffle=True)
wildfire_loader_val = DataLoader(splitter.val, batch_size=64, shuffle=True)
wildfire_loader_test = DataLoader(splitter.test, batch_size=64, shuffle=True)
```

Each dataloader will yield the image(transformed if requested) and the two targets (_fire_ and _clf\_confidence_)

For more example, please browse the tests

Any feedback is welcome

### CODE REVIEW

**Review by MateoLostanlen — COMMENTED** [review]
You made a mistake in your example, it is ->
from torch.utils.data import DataLoader
from pyronear.datasets.wildfire import (WildFireDataset,
                                        WildFireSplitter)

wildfire = WildFireDataset(metadata='wildfire.csv',
                           path_to_frames=path_to_frames)

ratios = {'train': 0.7, 'val': 0.15, 'test':0.15}

splitter =  WildFireSplitter(ratios)
splitter.fit(wildfire)

wildfire_loader_train = DataLoader(splitter.train, batch_size=64, shuffle=True)
wildfire_loader_val = DataLoader(splitter.val, batch_size=64, shuffle=True)
wildfire_loader_test = DataLoader(splitter.test, batch_size=64, shuffle=True)

**Review by MateoLostanlen — COMMENTED** [review]
Maybe my question is dumb, but I don't understand why you use "from skimage import io" and not pil to load the images. I have a problem to apply the torch vision transform afterwards (https://pytorch.org/docs/stable/torchvision/transforms.html). What do you use as a transform then ?

**Comment by x0s** [comment]
> You made a mistake in your example, it is ->
> from torch.utils.data import DataLoader
> from pyronear.datasets.wildfire import (WildFireDataset,
> WildFireSplitter)
> 
> wildfire = WildFireDataset(metadata='wildfire.csv',
> path_to_frames=path_to_frames)
> 
> ratios = {'train': 0.7, 'val': 0.15, 'test':0.15}
> 
> splitter = WildFireSplitter(ratios)
> splitter.fit(wildfire)
> 
> wildfire_loader_train = DataLoader(splitter.train, batch_size=64, shuffle=True)
> wildfire_loader_val = DataLoader(splitter.val, batch_size=64, shuffle=True)
> wildfire_loader_test = DataLoader(splitter.test, batch_size=64, shuffle=True)

Thanks for your feedback!
Indeed, I forget to update the description.
So, If I understand it right, only these two lines needs to be updated:
```python
splitter = WildFireSplitter(ratios)
splitter.fit(wildfire)
```

**Review by MateoLostanlen — APPROVED** [review]
